api/v1alpha2/linodecluster_types.go

```diff
@@ -111,7 +111,7 @@ type NetworkSpec struct {
 	// Ignored if the LoadBalancerType is set to anything other than dns
 	// If not set, defaults linode dns
 	// +optional
-	DNSProvider int `json:"dnsProvider,omitempty"`
+	DNSProvider string `json:"dnsProvider,omitempty"`
```

Comment:
do want this to be an enum? - and does switching types need a CRD migration

Reply:
unsure about the need for migration but I've added enum kubebuilder check to restrict this to linode/akamai